packages/headless/src/controllers/result-list/headless-result-selection-helpers.ts

```diff
@@ -0,0 +1,62 @@
+import {Result} from '../../api/search/search/result';
+import {Engine} from '../../app/headless-engine';
+import {ResultAnalyticsActions} from '../../features/analytics';
+
+// 1 second is a reasonable amount of time to catch most longpress actions
+const defaultDelay = 1000;
+
+/**
+ * Helpers that more easily allow sending analytics when a search result is selected.
+ */
+export type ResultSelectionHelpers = ReturnType<
+  typeof buildResultSelectionHelpers
+>;
+
+/**
+ * Creates helpers that more easily allow sending analytics when a search result is selected.
+ */
+export function buildResultSelectionHelpers(engine: Engine, result: Result) {
```

Comment:
It's really important that all our controllers follow a predictable pattern. Here the second argument must be `props` to stay consistent, which has a property called `options` that holds the `result` (required), and a `delay` optional.

```
props : {
    options: {
        result: Result,
        delay: number;
    }
}
```

If it is not already the case, I would keep the result somewhat loosely typed since every client's result may differ.